soc/riscv/openisa_rv32m1/soc.c

```diff
@@ -189,6 +189,27 @@ static void rv32m1_switch_to_sirc(void)
 	rv32m1_switch_sys_clk(&sys_clk_config_sirc);
 }
 
+/**
+ * @brief Setup peripheral clocks
+ *
+ * Setup the peripheral clock sources.
+ */
+static void rv32m1_setup_peripheral_clocks(void)
+{
+#ifdef DT_OPENISA_RV32M1_TPM_PWM_0_BASE_ADDRESS
+	CLOCK_SetIpSrc(kCLOCK_Tpm0, kCLOCK_IpSrcFircAsync);
```

Comment:
Other rv32m1 drivers (i2c, uart, spi) do this in the driver. Is there a reason you did this here instead?

Reply:
I would like to handle the PCC IP clock sources on the RV32M1 similar to how we are handling the PCC on the NXP KE1xF SoC series (move all calls to `CLOCK_SetIpSrc()` to `soc.c` and allow setting the source clocks in the device tree).

I will convert the remaining RV32M1 drivers if/when this PR is accepted.